internal/pkg/fleetapi/action.go

```diff
@@ -27,13 +31,43 @@ const (
 	ActionTypeSettings = "SETTINGS"
 	// ActionTypeInputAction specifies agent action.
 	ActionTypeInputAction = "INPUT_ACTION"
+	// ActionTypeCancel specifies a cancel action.
+	ActionTypeCancel = "CANCEL"
+)
+
+// Error values that the Action interface can return
+var (
+	ErrNoStartTime  = fmt.Errorf("action has no start time")
+	ErrNoExpiration = fmt.Errorf("action has no expiration")
 )
 
 // Action base interface for all the implemented action from the fleet API.
 type Action interface {
 	fmt.Stringer
 	Type() string
 	ID() string
+	// StartTime returns the earliest time an action should start (for schduled actions)
+	// Only ActionUpgrade implements this at the moment
+	StartTime() (time.Time, error)
+	// Expiration returns the time where an action is expired and should not be ran (for scheduled actions)
+	// Only ActionUpgrade implements this at the moment
+	Expiration() (time.Time, error)
+}
+
+// FleetAction represents an action from fleet-server.
+// should copy the action definition in fleet-server/model/schema.json
+type FleetAction struct {
```

Comment:
I wanted to separate what we receive from fleet as far as check ins go instead of using `ActionApp` to better delineate concerns